### PR TITLE
BUGFIX: Vetext two way lambda parsing bugfix

### PR DESCRIPTION
**Pull request by jakehova**
Bug fix to resolve issue with deserialization mms messages from twilio.  This required changes to how an alb event processes messages (namely uses dict instead of of json.loads and removes AddOns key/value from object) and sqs processes (uses json.loads).

Also added the unit testing file that I was using to test the lambda locally.

### CODE REVIEW

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 1]
```diff
@@ -127,27 +136,27 @@ def read_from_ssm(key: str) -> str:
...
     )
 
-    logger.info(response)
-
     return response.get("Parameter", {}).get("Value", '')
 
 def make_vetext_request(request_body):    
+    # We have been directed by the VeText team to ignore SSL validation
+    #   that is why we use the ssl._create_unverified_context method
     connection = http.client.HTTPSConnection(os.environ.get('vetext_api_endpoint_domain'),  context = ssl._create_unverified_context())
```
Comment (kalbfled) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:144]: I know this isn't a current change, but . . .
```
connection = http.client.HTTPSConnection(os.getenv('vetext_api_endpoint_domain'),  context = ssl._create_unverified_context())
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 2]
```diff
@@ -156,22 +165,31 @@ def make_vetext_request(request_body):
...
 
-def push_to_sqs(event):
-    """Places event on queue to be retried at a later time"""
+def push_to_sqs(event_body):
+    """Places event body dictionary on queue to be retried at a later time"""
+    logger.info("Placing event_body on retry queue")
+
     sqs = boto3.client('sqs')
     queue_url = os.environ.get('vetext_request_drop_sqs_url')
+    logger.debug(f"Retrieved queue_url: {queue_url}")
```
Comment (kalbfled) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:190]: ditto C string formatting syntax

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 3]
```diff
@@ -156,22 +165,31 @@ def make_vetext_request(request_body):
...
     response = connection.getresponse()
+    
+    logger.info(f"VeText call complete, with response: {str(response.status)}")
+    logger.debug(response.read().decode())
 
     return response    
 
-def push_to_sqs(event):
-    """Places event on queue to be retried at a later time"""
+def push_to_sqs(event_body):
```
Comment (kalbfled) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:184]: I recommend returning a boolean indicating if the boto call succeeded.
```
def push_to_sqs(event_body: dict) -> bool:
```

File: tests/lambda_functions/vetext_incoming_forwarder_lambda/test_vetext_incoming_forwarder_lambda.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+from lambda_functions.vetext_incoming_forwarder_lambda.vetext_incoming_forwarder_lambda import vetext_incoming_forwarder_lambda_handler, process_body_from_alb_invocation
+import pytest
+from unittest import mock
```
Comment (kalbfled) [inline tests/lambda_functions/vetext_incoming_forwarder_lambda/test_vetext_incoming_forwarder_lambda.py:3]: Do you need this?  You used the mocker fixture in your test functions.

File: tests/lambda_functions/vetext_incoming_forwarder_lambda/test_vetext_incoming_forwarder_lambda.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
...S0sqeX1INTZQDCl7nOqNbPjqx0FKgDzg=', 'body': '{"ToCountry": "US", "SmsMessageSid": "SM031265e1d5c5e56c6243509a1eafc7e8", "NumMedia": "0", "FromZip": "74344", "SmsSid": "SM031265e1d5c5e56c6243509a1eafc7e8", "FromState": "OK", "SmsStatus": "received", "FromCity": "GROVE", "Body": "Y1", "FromCountry": "US", "To": "53079", "MessagingServiceSid": "MGbe6bb27a9ca15de276c5b86f0d9e9c2e", "NumSegments": "1", "ReferralNumMedia": "0", "MessageSid": "SM031265e1d5c5e56c6243509a1eafc7e8", "AccountSid": "AC551fc6086f93a83492466fc604c68af4", "From": "+19188015777", "ApiVersion": "2010-04-01"}', 'attributes': {'ApproximateReceiveCount': '1', 'SentTimestamp': '1658165878744', 'SenderId': 'AROASQBEVZJB7JNKLLQIY:project-prod-vetext-incoming-forwarder-lambda', 'ApproximateFirstReceiveTimestamp': '1658245723375'}, 'messageAttributes': {'source': {'stringValue': 'twilio', 'stringListValues': [], 'binaryListValues': [], 'dataType': 'String'}}, 'md5OfMessageAttributes': '54818a2ef8e5363fbdb8d318121ef7c4', 'md5OfBody': '63d3b9430c4741f41a20554ed655db1f', 'eventSource': 'aws:sqs', 'eventSourceARN': 'arn:aws-us-gov:sqs:us-gov-west-1:171875617347:prod-vetext-failed-request-queue', 'awsRegion': 'us-gov-west-1'}]}
```
Comment (kalbfled) [inline tests/lambda_functions/vetext_incoming_forwarder_lambda/test_vetext_incoming_forwarder_lambda.py:8]: I would move these to another file, format them, and import them into this test.

File: tests/lambda_functions/vetext_incoming_forwarder_lambda/test_vetext_incoming_forwarder_lambda.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  assert response['statusCode'] == 500
+  sqs_mock.assert_not_called()
+
+@pytest.mark.parametrize('event', [(albInvokedWithoutAddOn), (albInvokeWithAddOn)])
+def test_failed_alb_invocation_call_throws_key_error_exception(mocker, event):
+  sqs_mock = mocker.patch('vetext_incoming_forwarder_lambda.push_to_sqs')
+  mocker.patch('vetext_incoming_forwarder_lambda.process_body_from_alb_invocation', side_effect=KeyError)
+  response = vetext_incoming_forwarder_lambda_handler(event, None)
+  
+  assert response['statusCode'] == 424
```
Comment (kalbfled) [inline tests/lambda_functions/vetext_incoming_forwarder_lambda/test_vetext_incoming_forwarder_lambda.py:88]: ditto "failed dependency" ?

**Review by kalbfled — CHANGES_REQUESTED** [review]

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 7]
```diff
@@ -96,13 +98,11 @@ def process_body_from_sqs_invocation(event):
...
         # parse_qsl converts url-encoded strings to array of tuple objects
         # event_body takes the array of tuples and creates a dictionary
-        try:
-            event_body_decoded = parse_qsl(b64decode(record["body"]).decode('utf-8'))
-            event_body = dict(event_body_decoded)
-            event_bodies.append(event_body)
-        except:
-            push_to_sqs(record["body"])
-
+        logger.debug(f"Processing record body from SQS: {record['body']}")
```
[the remark this answers is not in the document]
Comment (jakehova) [reply lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:101]: Updated this to: 

```
event_body = record.get("body", "")

      if (event_body == ""):
          logger.info("event_body from sqs record was not present")
          logger.debug(record)
          continue

      logger.debug("Processing record body from SQS: %s", event_body)
      event_body = json.loads(record["body"])
      logger.info("Successfully converted record body from sqs to json")
      event_bodies.append(event_body)
```

There really shouldn't be an instance where body is empty or not present but if it is, then I'm just making note of it and moving to the next record in thee loop

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py (hunk 2 above)
Comment (jakehova) [reply lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:190]: Changed to: 
```
logger.debug("Retrieved queue_url: %s", queue_url)
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 8]
```diff
@@ -96,28 +95,49 @@ def process_body_from_sqs_invocation(event):
...
-            event_body_decoded = parse_qsl(b64decode(record["body"]).decode('utf-8'))
-            event_body = dict(event_body_decoded)
-            event_bodies.append(event_body)
-        except:
-            push_to_sqs(record["body"])
+        try: 
+            event_body = record.get("body", "")
 
+            if (event_body == ""):
+                logger.info("event_body from sqs record was not present")
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:102]:
```suggestion
                logger.info("event body from sqs record was not present")
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 9]
```diff
@@ -96,28 +95,49 @@ def process_body_from_sqs_invocation(event):
...
+                logger.debug(record)
+                continue
+
+            logger.debug("Processing record body from SQS: %s", event_body)
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:106]:
```suggestion
            logger.debug(f"Processing record body from SQS: {event_body}")
```

Comment (k-macmillan) [reply lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:106]: I hope to never see a %s in python code ever again.

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 10]
```diff
@@ -96,28 +95,49 @@ def process_body_from_sqs_invocation(event):
...
+    event_body_encoded = event.get("body", "")
 
-    return event_bodies
-    
+    if (event_body_encoded == ""):
+        logger.info("event_body from alb record was not present")
+        logger.debug(event)        
+
+    event_body_decoded = parse_qsl(b64decode(event_body_encoded).decode('utf-8'))
+    logger.info(f"Decoded event_body {event_body_decoded}")
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:130]:
```suggestion
    logger.info(f"Decoded event body: {event_body_decoded}")
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 11]
```diff
@@ -96,28 +95,49 @@ def process_body_from_sqs_invocation(event):
...
+
+    event_body = dict(event_body_decoded)
+    logger.info("Converted body to dictionary")
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:133]:
```suggestion
    logger.info("Converted body to dictionary")
    logger.debug(f"Converted body dictionary: {event_body}")
```

Comment (k-macmillan) [reply lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:133]: The body is more of a debug thing, and having that could help us to narrow down issues.

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 12]
```diff
@@ -96,28 +95,49 @@ def process_body_from_sqs_invocation(event):
...
+
+    if 'AddOns' in event_body:        
+        logger.info(f"AddOns present in event_body: {event_body['AddOns']}")
+        del event_body['AddOns']
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:137]:
```suggestion
        event_body.pop('AddOns', None)
```

Comment (k-macmillan) [reply lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:137]: Del will result in a key error if it's not there. Use [pop](https://docs.python.org/3/library/stdtypes.html#dict.pop).

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 13]
```diff
@@ -156,30 +176,50 @@ def make_vetext_request(request_body):
 
     json_data = json.dumps(body)
 
+    logger.info("Making POST Request to VeText using: " + os.getenv('vetext_api_endpoint_domain') + os.getenv('vetext_api_endpoint_path'))
+    logger.debug(json_data)
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:180]:
```suggestion
    logger.debug(f"json dumps: {json_data}")
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 14]
```diff
@@ -156,30 +176,50 @@ def make_vetext_request(request_body):
...
-        os.environ.get('vetext_api_endpoint_path'),
+        os.getenv('vetext_api_endpoint_path'),
         json_data,
         headers)
-
+    
     response = connection.getresponse()
+    
+    logger.info("VeText call complete with response: %d", response.status)
+    logger.debug(response.read().decode())
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:191]:
```suggestion
    logger.debug(f"VEText response: {response}")
```
We don't want to run read/decode on unverified data.

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 15]
```diff
@@ -156,30 +176,50 @@ def make_vetext_request(request_body):
...
         'POST',
-        os.environ.get('vetext_api_endpoint_path'),
+        os.getenv('vetext_api_endpoint_path'),
         json_data,
         headers)
-
+    
     response = connection.getresponse()
+    
+    logger.info("VeText call complete with response: %d", response.status)
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:190]:
```suggestion
    logger.info(f"VeText call complete with response: {response.status}")
```
No %s in python strings please unless necessary. Use f-strings.

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 16]
```diff
@@ -156,30 +176,50 @@ def make_vetext_request(request_body):
...
 
-    queue_msg = json.dumps(event)
-    queue_msg_attrs = {
-        'source': {
-            'DataType': 'String',
-            'StringValue': 'twilio'
+    try:
+        sqs = boto3.client('sqs')
+        queue_url = os.getenv('vetext_request_drop_sqs_url')
+        logger.debug("Retrieved queue_url: %s", queue_url)
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:202]:
```suggestion
        logger.debug(f"Retrieved queue_url: {queue_url}")
```

**Review by k-macmillan — CHANGES_REQUESTED** [review]

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 17]
```diff
@@ -96,28 +95,49 @@ def process_body_from_sqs_invocation(event):
...
+    event_body_encoded = event.get("body", "")
 
-    return event_bodies
-    
+    if (event_body_encoded == ""):
+        logger.info("event_body from alb record was not present")
+        logger.debug(event)        
+
+    event_body_decoded = parse_qsl(b64decode(event_body_encoded).decode('utf-8'))
+    logger.info(f"Decoded event body {event_body_decoded}")
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:130]:
```suggestion
    logger.debug(f"Decoded event body {event_body_decoded}")
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 18]
```diff
@@ -96,28 +95,49 @@ def process_body_from_sqs_invocation(event):
...
+
+    event_body = dict(event_body_decoded)
+    logger.info(f"Converted body to dictionary: {event_body}")
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:133]:
```suggestion
    logger.debug(f"Converted body to dictionary: {event_body}")
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 19]
```diff
@@ -156,30 +176,50 @@ def make_vetext_request(request_body):
...
 
     return response    
 
-def push_to_sqs(event):
-    """Places event on queue to be retried at a later time"""
-    sqs = boto3.client('sqs')
-    queue_url = os.environ.get('vetext_request_drop_sqs_url')
+def push_to_sqs(event_body) -> bool:
+    """Places event body dictionary on queue to be retried at a later time"""
+    logger.info("Placing event_body on retry queue")
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:197]:
```suggestion
    logger.info("Placing event_body on retry queue")
    logger.debug(f"Preparing for SQS: {event_body}")
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 20]
```diff
@@ -34,54 +33,54 @@ def vetext_incoming_forwarder_lambda_handler(event: any, context: any):
...
+
+                if response.status != 200:
+                    logger.info("VeText call failed. Moving event body to failover queue")
+                    push_to_sqs(event_body)
+                
+                responses.append(response)
+            except http.client.HTTPException as e:
+                logger.info("HttpException With Call To VeText")                
+                logger.exception(e)                             
+                logger.debug(event_body)
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:60]:
```suggestion
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 21]
```diff
@@ -34,54 +33,54 @@ def vetext_incoming_forwarder_lambda_handler(event: any, context: any):
...
+                responses.append(response)
+            except http.client.HTTPException as e:
+                logger.info("HttpException With Call To VeText")                
+                logger.exception(e)                             
+                logger.debug(event_body)   
+                push_to_sqs(event_body)
+            except Exception as e:
+                logger.info("General Exception With Call to VeText")                
+                logger.exception(e)                        
+                logger.debug(event_body)
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:65]:
```suggestion
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 22]
```diff
@@ -156,30 +176,50 @@ def make_vetext_request(request_body):
...
-    sqs.send_message(QueueUrl=queue_url,
-                    MessageAttributes=queue_msg_attrs,
-                    MessageBody=queue_msg)
+        sqs.send_message(QueueUrl=queue_url,
+                        MessageAttributes=queue_msg_attrs,
+                        MessageBody=queue_msg)
+        
+        logger.info("Completed enqueue of message to retry queue")
+
+        return True
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:218]:
```suggestion
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 23]
```diff
@@ -156,30 +176,50 @@ def make_vetext_request(request_body):
...
+    except Exception as e:
+        logger.info("Push to SQS Exception")
+        logger.info(event_body)
+        logger.exception(e)        
+        
+        return False
```
Comment (k-macmillan) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:224]:
```suggestion
```

**Review by k-macmillan — CHANGES_REQUESTED** [review]

**Review by k-macmillan — CHANGES_REQUESTED** [review]

**Comment by k-macmillan** [comment]
Approving so the process isn't held up but this needs to be tested in Dev prior to a merge.

**Review by k-macmillan — DISMISSED** [review]

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 24]
```diff
@@ -96,28 +93,49 @@ def process_body_from_sqs_invocation(event):
...
-        try:
-            event_body_decoded = parse_qsl(b64decode(record["body"]).decode('utf-8'))
-            event_body = dict(event_body_decoded)
-            event_bodies.append(event_body)
-        except:
-            push_to_sqs(record["body"])
+        try: 
+            event_body = record.get("body", "")
+
+            if (event_body == ""):
```
Comment (kalbfled) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:99]:
```
if not event_body:
```

https://pep8.org/#string-quotes  <-- search for "empty"

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 25]
```diff
@@ -96,28 +93,49 @@ def process_body_from_sqs_invocation(event):
...
     # event["body"] is a base 64 encoded string
     # parse_qsl converts url-encoded strings to array of tuple objects
     # event_body takes the array of tuples and creates a dictionary
-    event_body_decoded = parse_qsl(b64decode(event["body"]).decode('utf-8'))
-    event_bodies.append(dict(event_body_decoded))
+    event_body_encoded = event.get("body", "")
 
-    return event_bodies
-    
+    if (event_body_encoded == ""):
```
Comment (kalbfled) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:123]: Ditto.  This is a good place to return None.

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 26]
```diff
@@ -156,30 +174,47 @@ def make_vetext_request(request_body):
...
+    logger.debug(f"json dumps: {json_data}")
+    
     connection.request(
         'POST',
-        os.environ.get('vetext_api_endpoint_path'),
+        os.getenv('vetext_api_endpoint_path'),
         json_data,
         headers)
-
+
```
Comment (kalbfled) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:185]: remove

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py [hunk 27]
```diff
@@ -34,54 +33,52 @@ def vetext_incoming_forwarder_lambda_handler(event: any, context: any):
...
+                logger.exception(e)                                        
+                push_to_sqs(event_body)
 
         logger.debug(responses)
         
         return {
             'statusCode': 200
         }
-    except KeyError as e:
+    except KeyError as e:
```
Comment (kalbfled) [inline lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py:71]: remove

File: tests/lambda_functions/vetext_incoming_forwarder_lambda/test_vetext_incoming_forwarder_lambda.py [hunk 28]
```diff
@@ -0,0 +1,105 @@
...
+
+
+lambda_module = "lambda_functions.vetext_incoming_forwarder_lambda.vetext_incoming_forwarder_lambda"
+
+
+@pytest.mark.parametrize('event', [(albInvokedWithoutAddOn), (albInvokeWithAddOn)])
+def test_verify_parsing_of_twilio_message(event):
+    response = process_body_from_alb_invocation(event)
+
+    assert len(response) > 0
```
Comment (kalbfled) [inline tests/lambda_functions/vetext_incoming_forwarder_lambda/test_vetext_incoming_forwarder_lambda.py:35]:
```
assert response, "The dictionary shouldn't be empty."
```
same pep8 reference as above

**Review by kalbfled — CHANGES_REQUESTED** [review]
I like how you've parameterized tests.

I won't hold you up if you choose not to refactor to catch exceptions locally and return None or boolean values in functions, rather than letting exceptions bubble up and get caught at the top level, but I want to iterate that such changes would improve the execution flow and readability significantly.

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py (hunk 27 above)
Comment (kalbfled) [reply lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py]: It still shows that you have excess white space at the end.

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py (hunk 25 above)
Comment (kalbfled) [reply lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py]:
```
if not even_body_encoded:
```

File: lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py (hunk 26 above)
Comment (kalbfled) [reply lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py]: still shows white space

**Review by kalbfled — APPROVED** [review]

**Review by k-macmillan — APPROVED** [review]

**Pull request merged**